openstf/model/train.py

```diff
@@ -144,14 +204,37 @@ def train_model_for_specific_pj(
     train_data, validation_data, test_data = split_data_train_validation_test(
         clean_data_with_features
     )
+    return {
+        "train_data": train_data,
+        "validation_data": validation_data,
+        "test_data": test_data,
+    }
 
-    context.perf_meter.checkpoint("preprocessing")
 
+def train_for_specific_pj(model_trainer, split_input_data):
+    """Train model.
+
+    Args:
+        model_trainer (MLModelType): model trainer
+        split_input_data (dict): dict of cleaned data with features, split in train,
+            validation and test
+    """
     # Train model
-    model_trainer.train(train_data, validation_data)
+    model_trainer.train(split_input_data.train_data, split_input_data.validation_data)
 
-    context.perf_meter.checkpoint("training")
 
+def predict_after_training_for_specific_pj(pj, model_trainer, split_input_data):
+    """Predict data with new trained model for model evaluation.
```

Comment:
Why would you add for model evaluation here. This function does not need to know such things I think?

Reply:
I was not sure if the function name was clear enough.